flow/shared/context.go

```diff
@@ -0,0 +1,83 @@
+package shared
+
+import (
+	"context"
+
+	"github.com/PeerDB-io/peerdb/flow/generated/protos"
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	FlowMetadataKey = "x-peerdb-flow-metadata"
+)
+
+type FlowMetadata struct {
+	FlowName    string
```

Comment:
will modifications to this struct cause non determinism errors? would using protobuf for encoding/decoding payload help versioning issues?

Reply:
I will test this, but I assume that it should not cause issues as headers should not be considered inputs